mlir: fatal error: `'mlir/Dialect/Mesh/Interfaces/ShardingInterface.h.inc' file not found`

Recent regression on Linux amd64:

```
FAILED: tools/mlir/lib/Dialect/Linalg/IR/CMakeFiles/obj.MLIRLinalgDialect.dir/LinalgDialect.cpp.o
/opt/sccache//sccache /build/source/build-llvm/bin/clang++ -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/build-llvm/tools/clang/stage2-bins/tools/mlir/lib/Dialect/Linalg/IR -I/build/source/mlir/lib/Dialect/Linalg/IR -I/build/source/build-llvm/tools/clang/stage2-bins/include -I/build/source/llvm/include -I/build/source/mlir/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/mlir/include -fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=../../../../ -ffile-prefix-map=/build/source/= -no-canonical-prefixes -Wundef -Werror=mismatched-tags -Werror=global-constructors -O2 -DNDEBUG -g1 -std=c++17  -fno-exceptions -funwind-tables -MD -MT tools/mlir/lib/Dialect/Linalg/IR/CMakeFiles/obj.MLIRLinalgDialect.dir/LinalgDialect.cpp.o -MF tools/mlir/lib/Dialect/Linalg/IR/CMakeFiles/obj.MLIRLinalgDialect.dir/LinalgDialect.cpp.o.d -o tools/mlir/lib/Dialect/Linalg/IR/CMakeFiles/obj.MLIRLinalgDialect.dir/LinalgDialect.cpp.o -c /build/source/mlir/lib/Dialect/Linalg/IR/LinalgDialect.cpp
In file included from /build/source/mlir/lib/Dialect/Linalg/IR/LinalgDialect.cpp:19:
/build/source/mlir/include/mlir/Dialect/Mesh/Interfaces/ShardingInterface.h:84:10: fatal error: 'mlir/Dialect/Mesh/Interfaces/ShardingInterface.h.inc' file not found
   84 | #include "mlir/Dialect/Mesh/Interfaces/ShardingInterface.h.inc"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/mlir/lib/Dialect/Linalg/IR/CMakeLists.txt
+++ b/mlir/lib/Dialect/Linalg/IR/CMakeLists.txt
@@ -13,7 +13,7 @@ add_mlir_dialect_library(MLIRLinalgDialect
   MLIRLinalgOpsEnumsIncGen
   MLIRLinalgOpsIncGen
   MLIRLinalgStructuredOpsIncGen
-  MLIRShardingInterface
+  MLIRShardingInterfaceIncGen
 
   LINK_LIBS PUBLIC
   MLIRAffineDialect

--- a/mlir/lib/Dialect/Linalg/IR/CMakeLists.txt
+++ b/mlir/lib/Dialect/Linalg/IR/CMakeLists.txt
@@ -13,6 +13,7 @@ add_mlir_dialect_library(MLIRLinalgDialect
   MLIRLinalgOpsEnumsIncGen
   MLIRLinalgOpsIncGen
   MLIRLinalgStructuredOpsIncGen
+  MLIRShardingInterface
 
   LINK_LIBS PUBLIC
   MLIRAffineDialect

